app/src/main/java/mod/tyron/compiler/IncrementalJavaCompiler.java

```diff
@@ -235,8 +352,110 @@ public CompilerOutputStream(StringBuffer buffer) {
+        String dexsDirectoryPath = new File(l, "dexs").getAbsolutePath();
+        String libsDirectoryPath = new File(l, "libs").getAbsolutePath();
+        String testkeyDirectoryPath = new File(l, "testkey").getAbsolutePath();
+        if (a(m + File.separator + "android.jar.zip", androidJarPath)) {
+            /* Delete android.jar */
+            g.c(l.getAbsolutePath() + File.separator + "android.jar");
+            new KB().a(androidJarPath, l.getAbsolutePath());
+        }
+        if (a(m + File.separator + "dexs.zip", dexsArchivePath)) {
+            g.b(dexsDirectoryPath);
+            g.f(dexsDirectoryPath);
+            new KB().a(dexsArchivePath, dexsDirectoryPath);
+        }
+        if (a(m + File.separator + "libs.zip", libsArchivePath)) {
+            g.b(libsDirectoryPath);
+            g.f(libsDirectoryPath);
+            new KB().a(libsArchivePath, libsDirectoryPath);
+        }
+        String jdkArchivePathInAssets = m + File.separator + "jdk.zip";
+        String jdkArchivePath = new File(l, "jdk.zip").getAbsolutePath();
+        /* Check if file size has changed */
+        if (a(jdkArchivePathInAssets, jdkArchivePath)) {
+            String jdkDirectoryPath = new File(l, "jdk").getAbsolutePath();
+            /* Delete the directory? */
+            g.b(jdkDirectoryPath);
+            /* Create the directories? */
+            g.f(jdkDirectoryPath);
+            /* Extract the archive to the directory? */
+            new KB().a(jdkArchivePath, jdkDirectoryPath);
+        }
+        String testkeyArchivePathInAssets = m + File.separator + "testkey.zip";
+        String testkeyArchivePath = new File(l, "testkey.zip").getAbsolutePath();
+        if (a(testkeyArchivePathInAssets, testkeyArchivePath)) {
+            /* We need to copy testkey.zip to filesDir */
+            g.b(testkeyDirectoryPath);
+            g.f(testkeyDirectoryPath);
+            new KB().a(testkeyArchivePath, testkeyDirectoryPath);
+
+        }
+
+        if (projectConfig.N.g) {
+            n.a("appcompat-1.0.0");
+            n.a("coordinatorlayout-1.0.0");
+            n.a("material-1.0.0");
+        }
+        if (projectConfig.N.h) {
+            n.a("firebase-common-19.0.0");
+        }
+        if (projectConfig.N.i) {
+            n.a("firebase-auth-19.0.0");
+        }
+        if (projectConfig.N.j) {
+            n.a("firebase-database-19.0.0");
+        }
+        if (projectConfig.N.k) {
+            n.a("firebase-storage-19.0.0");
+        }
+        if (projectConfig.N.m) {
+            n.a("play-services-maps-17.0.0");
+        }
+        if (projectConfig.N.l) {
+            n.a("play-services-ads-18.2.0");
+        }
+        if (projectConfig.N.o) {
+            n.a("gson-2.8.0");
+        }
+        if (projectConfig.N.n) {
+            n.a("glide-4.11.0");
+        }
+        if (projectConfig.N.p) {
+            n.a("okhttp-3.9.1");
+        }
+        ExtLibSelected.a(projectConfig.N.x, n);
+    }
```

Comment:
These methods seem to be copied from Dp.java. Could you use them instead, as you have a Dp object anyway?